blog/user_roles.md

```diff
@@ -0,0 +1,294 @@
+
+We used Pow! for our authentication and that library also has a guide for implementing a user roles system.
+Let's see how far that get's us and I'd like to actually take it a bit further and implement different
+sign up flows for these different user roles.
+
+But as always, we shouldn't get ahead of ourselves and make it too complicated too soon, so we'll first
+implement basic user roles. We are going to create these users in a simple manner by just adding a selection
+before the registration form that will set this field for the user. I already know I want to be able to define
+multple roles in the future so I'll take that into account when creating the field I'm going to use, but we'll
+not complicate things too much so 1 role can be selected when signing up.
+
+### Migration and schema
+
+We'll need to store the user roles, so we'll add a migration to store a list of users in the users table:
+
+```elixir
+defmodule StudentManager.Repo.Migrations.AddRolesToUsers do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :roles, {:array, :string}, default: ["student"]
+    end
+  end
+end
+```
+
+In this migration I'm taking advantage of the Postgres Array type to store the list of users in the db. It is
+of course also possible to store the roles in a different table and at those as an association, but for now
+this will do and we can just validate the allowed role values in the Changeset we'll use to store and update
+this user role info.
+
+We'll also need to add this new field to our User struct to be able to use it in our application. Here I have also added the validation of the user role values. 
+
+```elixir
+defmodule StudentManager.Accounts.User do
+  use Ecto.Schema
+  use Pow.Ecto.Schema
+
+  schema "users" do
+    field(:roles, {:array, :string}, default: ["student"])
+
+    pow_user_fields()
+
+    timestamps()
+  end
+
+  def changeset_role(user_or_changeset, attrs) do
+    user_or_changeset
+    |> Changeset.cast(attrs, [:roles])
+    |> Changeset.validate_inclusion(:roles, ~w(student teacher))
+  end
+end
+```
+
+### Testing
+
+In the first blog post I didn;t mention testing at all, but it is a very important part of building a reliable app that is easy to refactor and change with confidence.
+Without going into too much details about a proper and full-fledged test setup (I'll cover that in a separate post in the future), I would like to test this
+user role addition just to check if the above holds up and we are actually able to add these roles to a user. Now, when creating this app and setting this up in the
+first post I briefly mentioned that I was going to add the user under the accounts context. Now, we currently don't have any contexts setup, so let's take advantage
+of Phoenix generators to set it up after we've alread created the user. I have confidence that Dan has taken care of the basic testing of the basic user level
```

Comment:
Who's `Dan`? This whole section is perhaps a little hard to follow if you haven't (recently) read the previous post. Could you maybe reword it to make it easier to follow for people who haven't recently read the previous post?

Reply:
good point I have reworded and also linked to my previous post while I was at it :)